<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1073</width>
    <height>750</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item row="0" column="0">
     <layout class="QGridLayout" name="gridLayout">
      <property name="leftMargin">
       <number>6</number>
      </property>
      <property name="topMargin">
       <number>6</number>
      </property>
      <property name="rightMargin">
       <number>9</number>
      </property>
      <property name="bottomMargin">
       <number>6</number>
      </property>
      <item row="4" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Type</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>6</number>
        </property>
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QDoubleSpinBox" name="room_grav_x">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <double>-20.000000000000000</double>
          </property>
          <property name="maximum">
           <double>20.000000000000000</double>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="room_grav_y">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimum">
           <double>-20.000000000000000</double>
          </property>
          <property name="maximum">
           <double>20.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Room Properties</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_20">
        <property name="text">
         <string>Music</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Gravity</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QDoubleSpinBox" name="distance">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximum">
         <double>255.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="Permeability">
        <property name="text">
         <string>Permeability</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QSpinBox" name="permeability">
        <property name="minimum">
         <number>-1</number>
        </property>
        <property name="maximum">
         <number>100</number>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="room_type">
        <property name="maxCount">
         <number>16</number>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Distance</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Door Properties</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QComboBox" name="room_music">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="1">
     <layout class="QGridLayout" name="gridLayout_4">
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="1" column="0">
       <widget class="QScrollBar" name="horizontalScrollBar">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>1024</number>
        </property>
        <property name="pageStep">
         <number>4096</number>
        </property>
        <property name="value">
         <number>32767</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QScrollBar" name="verticalScrollBar">
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>1024</number>
        </property>
        <property name="pageStep">
         <number>4096</number>
        </property>
        <property name="value">
         <number>32767</number>
        </property>
        <property name="sliderPosition">
         <number>32767</number>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="GLViewWidget" name="viewWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>640</width>
          <height>480</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1073</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="fileNew"/>
    <addaction name="fileOpen"/>
    <addaction name="separator"/>
    <addaction name="fileSave"/>
    <addaction name="fileSaveAs"/>
    <addaction name="separator"/>
    <addaction name="fileLoadBackground"/>
    <addaction name="fileImportWalls"/>
    <addaction name="fileCAOS"/>
    <addaction name="separator"/>
    <addaction name="fileClose"/>
    <addaction name="appExit"/>
   </widget>
   <widget class="QMenu" name="menuEdit">
    <property name="title">
     <string>Edit</string>
    </property>
    <widget class="QMenu" name="menuZoom">
     <property name="title">
      <string>Zoom</string>
     </property>
     <addaction name="zoomIn"/>
     <addaction name="zoomOut"/>
     <addaction name="zoomActual"/>
    </widget>
    <widget class="QMenu" name="menuSelect_2">
     <property name="title">
      <string>Select</string>
     </property>
     <addaction name="selectAll"/>
     <addaction name="selectNone"/>
     <addaction name="selectInvert"/>
     <addaction name="selectConnected"/>
     <addaction name="selectOverlap"/>
     <addaction name="selectRoomType"/>
     <addaction name="selectRoomMusic"/>
    </widget>
    <addaction name="editUndo"/>
    <addaction name="editRedo"/>
    <addaction name="separator"/>
    <addaction name="editCut"/>
    <addaction name="editCopy"/>
    <addaction name="editPaste"/>
    <addaction name="editDelete"/>
    <addaction name="separator"/>
    <addaction name="menuZoom"/>
    <addaction name="menuSelect_2"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="viewChecker"/>
    <addaction name="viewRooms"/>
    <addaction name="viewHalls"/>
    <addaction name="viewLinks"/>
    <addaction name="separator"/>
    <addaction name="viewGravity"/>
   </widget>
   <widget class="QMenu" name="menuAbout">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="aboutHelp"/>
    <addaction name="aboutAbout"/>
    <addaction name="separator"/>
    <addaction name="aboutQt"/>
   </widget>
   <widget class="QMenu" name="menuLinks">
    <property name="title">
     <string>Links</string>
    </property>
    <addaction name="linkAdd"/>
    <addaction name="linkRemove"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="toolNew"/>
    <addaction name="toolFace"/>
    <addaction name="toolDuplicate"/>
    <addaction name="separator"/>
    <addaction name="toolReseat"/>
    <addaction name="separator"/>
    <addaction name="toolTranslate"/>
    <addaction name="toolRotate"/>
    <addaction name="toolScale"/>
    <addaction name="toolExtrude"/>
    <addaction name="toolSlice"/>
    <addaction name="separator"/>
    <addaction name="toolRealign"/>
    <addaction name="toolPropagate"/>
   </widget>
   <widget class="QMenu" name="menuDebug">
    <property name="title">
     <string>Debug</string>
    </property>
    <addaction name="debugTreeSymmetry"/>
    <addaction name="debugDoorSymmetry"/>
    <addaction name="separator"/>
    <addaction name="viewRoomId"/>
    <addaction name="debugSelectRoom"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuEdit"/>
   <addaction name="menuTools"/>
   <addaction name="menuLinks"/>
   <addaction name="menuView"/>
   <addaction name="menuDebug"/>
   <addaction name="menuAbout"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="sizeGripEnabled">
    <bool>false</bool>
   </property>
  </widget>
  <action name="fileNew">
   <property name="icon">
    <iconset theme="document-new">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="fileOpen">
   <property name="icon">
    <iconset theme="document-open">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="fileSaveAs">
   <property name="icon">
    <iconset theme="document-save-as">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save As</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="fileSave">
   <property name="icon">
    <iconset theme="document-save">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="fileClose">
   <property name="icon">
    <iconset theme="document-close">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Close</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+W</string>
   </property>
  </action>
  <action name="appExit">
   <property name="icon">
    <iconset theme="application-exit">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="editUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-undo">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Z</string>
   </property>
  </action>
  <action name="editRedo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-redo">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Redo</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+Z</string>
   </property>
  </action>
  <action name="selectAll">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="select">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Select All</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="zoomIn">
   <property name="icon">
    <iconset theme="zoom-in">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Zoom In</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+=</string>
   </property>
  </action>
  <action name="zoomOut">
   <property name="icon">
    <iconset theme="zoom-out">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Zoom Out</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+-</string>
   </property>
  </action>
  <action name="zoomActual">
   <property name="icon">
    <iconset theme="zoom">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Actual Size</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+=</string>
   </property>
  </action>
  <action name="selectNone">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Select None</string>
   </property>
  </action>
  <action name="editCut">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-cut">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Cut</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="editCopy">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-copy">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Copy</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="editPaste">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-paste">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Paste</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
  <action name="editDelete">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset theme="edit-delete">
     <normaloff>.</normaloff>.</iconset>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
   <property name="shortcut">
    <string>Del</string>
   </property>
  </action>
  <action name="viewChecker">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Transparency Texture</string>
   </property>
  </action>
  <action name="viewRooms">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Rooms</string>
   </property>
  </action>
  <action name="viewLinks">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Links</string>
   </property>
  </action>
  <action name="aboutQt">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="aboutAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="aboutHelp">
   <property name="text">
    <string>Help</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="linkAdd">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Add Link</string>
   </property>
  </action>
  <action name="linkRemove">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Remove Link</string>
   </property>
  </action>
  <action name="viewHalls">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Halls</string>
   </property>
  </action>
  <action name="toolNew">
   <property name="text">
    <string>Create Room (N)</string>
   </property>
   <property name="toolTip">
    <string>Create Room</string>
   </property>
  </action>
  <action name="toolExtrude">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Extrude (E)</string>
   </property>
   <property name="toolTip">
    <string>Extrude</string>
   </property>
  </action>
  <action name="toolSlice">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Slice</string>
   </property>
   <property name="toolTip">
    <string>Slice (Ctrl+R)</string>
   </property>
  </action>
  <action name="musicMute">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Mute</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="fileImportWalls">
   <property name="text">
    <string>Load Room Types</string>
   </property>
  </action>
  <action name="fileLoadBackground">
   <property name="text">
    <string>Load Background</string>
   </property>
  </action>
  <action name="musicLoad">
   <property name="text">
    <string>Load FMOD Bank</string>
   </property>
  </action>
  <action name="toolTranslate">
   <property name="text">
    <string>Translate (G)</string>
   </property>
  </action>
  <action name="toolRotate">
   <property name="text">
    <string>Rotate (R)</string>
   </property>
  </action>
  <action name="toolScale">
   <property name="text">
    <string>Scale (S)</string>
   </property>
  </action>
  <action name="fileSaveRooms">
   <property name="text">
    <string>Save Room System</string>
   </property>
  </action>
  <action name="toolReseat">
   <property name="text">
    <string>Reseat</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+H</string>
   </property>
  </action>
  <action name="fileCAOS">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Export CAOS</string>
   </property>
  </action>
  <action name="viewGravity">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Gravity Direction</string>
   </property>
  </action>
  <action name="toolDuplicate">
   <property name="text">
    <string>Duplicate</string>
   </property>
   <property name="shortcut">
    <string>Shift+D</string>
   </property>
  </action>
  <action name="toolFace">
   <property name="text">
    <string>Add Face (F)</string>
   </property>
   <property name="shortcut">
    <string>F</string>
   </property>
  </action>
  <action name="toolRealign">
   <property name="text">
    <string>Align Gravity With Slice</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="toolPropagate">
   <property name="text">
    <string>Propagate Gravity</string>
   </property>
   <property name="shortcut">
    <string>Alt+G</string>
   </property>
  </action>
  <action name="selectInvert">
   <property name="text">
    <string>Invert Selection</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="selectConnected">
   <property name="text">
    <string>Connected</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="selectOverlap">
   <property name="text">
    <string>Overlapping</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+L</string>
   </property>
  </action>
  <action name="selectRoomType">
   <property name="text">
    <string>By Room Type</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+T</string>
   </property>
  </action>
  <action name="selectRoomMusic">
   <property name="text">
    <string>By Room Music</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+M</string>
   </property>
  </action>
  <action name="debugTreeSymmetry">
   <property name="text">
    <string>Test Tree Symmetry</string>
   </property>
  </action>
  <action name="debugDoorSymmetry">
   <property name="text">
    <string>Test Door Symmetry</string>
   </property>
  </action>
  <action name="viewRoomId">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>View Room IDs</string>
   </property>
  </action>
  <action name="debugSelectRoom">
   <property name="text">
    <string>Select Room by ID</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLViewWidget</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">src/glviewwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>fileClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>376</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
